vllm/distributed/parallel_state.py

```diff
@@ -26,6 +26,7 @@
 import pickle
 import weakref
 from collections import namedtuple
+from concurrent.futures import Future
```

Comment:
nit: we can do "import concurrent.futures" here. And explicitly say "def _to_concurrent_future(result: torch.Future) -> concurrent.futures.Future:" to help understand the output type.